src/modules/thirdParty/sagas.ts

```diff
@@ -110,4 +160,56 @@ export function* thirdPartySaga(builder: BuilderAPI) {
   function* handleBuyThirdPartyItemSlotSuccess() {
     yield put(closeModal('BuyItemSlotsModal'))
   }
+
+  function* handlePublishThirdPartyItemRequest(action: PublishThirdPartyItemsRequestAction) {
+    const { thirdParty, items } = action.payload
+    try {
+      const collectionId = items[0].collectionId!
+
+      const { signature, signedMessage, salt } = yield call(getPublishItemsSignature, thirdParty.id, items.length)
+
+      const { items: newItems, itemCurations: newItemCurations }: { items: Item[]; itemCurations: ItemCuration[] } = yield call(
+        [builder, 'publishCollection'],
+        collectionId,
+        items.map(i => i.id),
+        signedMessage,
+        signature,
+        items.length, // qty
+        salt
+      )
+
+      yield put(publishThirdPartyItemsSuccess(collectionId, newItems, newItemCurations))
+      yield put(closeModal('PublishThirdPartyCollectionModal'))
+    } catch (error) {
+      yield put(publishThirdPartyItemsFailure(error.message))
+    }
+  }
+
+  function* handlePushChangesThirdPartyItemRequest(action: PushChangesThirdPartyItemsRequestAction) {
+    try {
+      const { items } = action.payload
+      const collectionId = items[0].collectionId
+      if (!collectionId) return
+
+      const itemCurations: ItemCuration[] = yield select(getItemCurations, collectionId!)
+      const effects: CallEffect<void>[] = items.map(item => {
+        const curation = itemCurations.find(ic => ic.itemId === item.id)
```

Comment:
same comment as before but maybe we can rename this to itemCuration from ic for readability sake

Reply:
updated!